presto-native-execution/presto_cpp/main/SystemConnector.h

```diff
@@ -0,0 +1,233 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+
+#include "presto_cpp/main/SystemSplit.h"
+#include "presto_cpp/main/types/PrestoToVeloxConnector.h"
+
+#include "velox/connectors/Connector.h"
+
+namespace facebook::presto {
+
+class TaskManager;
+
+class SystemColumnHandle : public velox::connector::ColumnHandle {
+ public:
+  explicit SystemColumnHandle(const std::string& name) : name_(name) {}
+
+  const std::string& name() const {
+    return name_;
+  }
+
+ private:
+  const std::string name_;
+};
+
+class SystemTableHandle : public velox::connector::ConnectorTableHandle {
+ public:
+  explicit SystemTableHandle(
+      std::string connectorId,
```

Comment:
Do we usually prefer to pass by value and then move? Or pass by const reference and copy? When do we prefer one over the other?

Reply:
@arhimondr : Good question. I prefer pass by const ref and copy to avoid use after move at the caller. But I've seen pass by value and move as a common pattern in Velox as well.

Lets make this pass by const ref and copy to be safe.